extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleUtils.java

```diff
@@ -35,5 +46,103 @@ public static boolean isRetryable(Throwable t)
     return t instanceof IOException;
   }
 
-  public static final Predicate<Throwable> GOOGLE_RETRY = e -> isRetryable(e);
+  private static <T> T retryGoogleCloudStorageOperation(RetryUtils.Task<T> f) throws Exception
+  {
+    return RetryUtils.retry(f, GOOGLE_RETRY, RetryUtils.DEFAULT_MAX_TRIES);
+  }
+
+
+  public static Iterator<StorageObject> lazyFetchingStorageObjectsIterator(
```

Comment:
Doesn't have to be done now, but it may be worth building a common abstraction for this and the equivalent in S3Utils if we end up doing similar stuff for all the cloud object stores.

Reply:
I agree this would be nice to have, I will revisit this in a future PR, especially if another cloud (azure or whatever) has a similar API.